### ISSUE TITLE
`jit_compile=True` crashes on dead code involving `tf.argmin`/`tf.argmax` on empty-dimension tensors

### ISSUE THREAD

**Issue by wuyii8941** [tensorflow/tensorflow#117547]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf-nightly 2.22.0-dev20260503 (also tested on 2.21.0)

### Custom code

Yes

### Current Behavior

When a `tf.function` contains a `tf.argmin` or `tf.argmax` call on an empty-dimension tensor, and the result of that call is **not used in the return value** (dead code), `jit_compile=True` crashes with `tf2xla conversion failed`, while `jit_compile=False` runs correctly by eliminating the dead code.

This behavioral inconsistency means that switching from `jit_compile=False` to `jit_compile=True` can break code that previously worked, even when the problematic operation would never be executed at runtime.

### Expected Behavior

`jit_compile=True` should not crash on dead code. When an operation's result is not used in the function output, the XLA compilation pipeline should either:
1. Eliminate the dead code before attempting tf2xla conversion (matching the behavior of `jit_compile=False`), or
2. Handle `ArgMin`/`ArgMax` on empty-dimension tensors gracefully during tf2xla conversion, similar to how `reduce_max`/`reduce_sum` already handle this case.

The practical impact: code that works with `jit_compile=False` can break when switching to `jit_compile=True`, even though the problematic operation is unreachable. This matters for programs that conditionally produce empty tensors (e.g., filtering, masking, dynamic batching), where certain branches may legally contain operations on empty tensors.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

x = tf.zeros((4, 0), dtype=tf.float32)

def fn(x):
    dead = tf.cast(tf.argmin(x, axis=-1), tf.float32)  # dead code, result unused
    return tf.constant(42.0)

# Works: Grappler eliminates dead code before execution
print(tf.function(fn, jit_compile=False)(x))  # tf.Tensor(42.0, ...)

# Crashes: XLA tries to compile dead code and fails
print(tf.function(fn, jit_compile=True)(x))   # tf2xla conversion failed


The issue is specific to `tf.argmin` and `tf.argmax`. Other reductions (`reduce_max`, `reduce_sum`, `reduce_mean`) do not crash XLA even on empty-dimension tensors:


import tensorflow as tf

x = tf.zeros((4, 0), dtype=tf.float32)

for op_name, op in [("argmin", tf.argmin), ("argmax", tf.argmax),
                     ("reduce_max", tf.reduce_max), ("reduce_sum", tf.reduce_sum)]:
    def make_fn(body):
        def fn(x):
            dead = body(x, axis=-1)
            return tf.constant(42.0)
        return fn
    try:
        tf.function(make_fn(op), jit_compile=True)(x)
        print(f"{op_name}: OK")
    except:
        print(f"{op_name}: CRASH")

# argmin:     CRASH
# argmax:     CRASH
# reduce_max: OK
# reduce_sum: OK


### Relevant log output


Reduction axis -1 is empty in shape [4,0]

Stack trace for op definition: ...

  [[{{node ArgMin}}]]
tf2xla conversion failed while converting __inference_fn_26[_XlaMustCompile=true,...]
```

### Relevant log output

```shell

```

**Comment by avmohansai** [comment]
Hi @Venkat6871 ,
I Was able to reproduce the issue on Colab using `TF v2.21.0` and `TF-nightly` ,Please find the [gist](https://colab.sandbox.google.com/gist/avmohansai/2179e5e5f2db68e84b77f49aeffe9740/117547_tf_2-21-0_tf-nightly.ipynb) here for reference.
Thank you.
